src/main/atm2lndMod.F90

```diff
@@ -719,4 +739,222 @@ subroutine check_downscale_consistency(bounds, atm2lnd_inst, wateratm2lndbulk_in
 
   end subroutine check_downscale_consistency
 
+  subroutine downscale_hillslope_solar(bounds, atm2lnd_inst, surfalb_inst)
+    !
+    ! !DESCRIPTION:
+    ! Downscale incoming direct solar radiation based on local slope and aspect.
+    !
+    ! This is currently applied over columns
+    !
+    ! USES
+    use clm_varpar    , only : numrad
+
+    ! !ARGUMENTS:
+    type(bounds_type)  , intent(in)    :: bounds  
+    type(surfalb_type) , intent(in)    :: surfalb_inst
+    type(atm2lnd_type) , intent(inout) :: atm2lnd_inst
+    !
+    ! !LOCAL VARIABLES:
+    integer  :: c,l,g,n      ! indices
+    real(r8) :: norm(numrad)
+    real(r8) :: sum_solar(bounds%begg:bounds%endg,numrad)
+    real(r8) :: sum_wtlunit(bounds%begg:bounds%endg)
+
+    character(len=*), parameter :: subname = 'downscale_hillslope_solar'
+    !-----------------------------------------------------------------------
+    
+    associate(&
+         ! Gridcell-level fields:
+         forc_solai_grc  =>    atm2lnd_inst%forc_solai_grc , & ! Input:  [real(r8) (:)]  gridcell indirect incoming solar radiation
+         forc_solad_grc  =>    atm2lnd_inst%forc_solad_grc , & ! Input:  [real(r8) (:)]  gridcell direct incoming solar radiation
+         coszen_grc      =>    surfalb_inst%coszen_grc     , & ! Input:  [real(r8) (:)]  cosine of solar zenith angle            
+         
+         ! Column-level fields:
+         forc_solar_col  =>    atm2lnd_inst%forc_solar_col , & ! Output:  [real(r8) (:)]  column total incoming solar radiation
+         forc_solad_col  =>    atm2lnd_inst%forc_solad_col , & ! Output:  [real(r8) (:)]  column direct incoming solar radiation
+         coszen_col      =>    surfalb_inst%coszen_col       & ! Input:   [real(r8) (:)]  cosine of solar zenith angle            
+         )
+      
+      ! Initialize column forcing
+      sum_solar(bounds%begg:bounds%endg,1:numrad) = 0._r8
+      sum_wtlunit(bounds%begg:bounds%endg) = 0._r8
+      do c = bounds%begc,bounds%endc
+         g = col%gridcell(c)
+         forc_solad_col(c,1:numrad)  = forc_solad_grc(g,1:numrad)
+         if (col%is_hillslope_column(c)) then
+            if (coszen_grc(g) > 0._r8) then
+               forc_solad_col(c,1:numrad)  = forc_solad_grc(g,1:numrad)*(coszen_col(c)/coszen_grc(g))
+            endif
+            
+            sum_solar(g,1:numrad) = sum_solar(g,1:numrad) + col%wtlunit(c)*forc_solad_col(c,1:numrad)
+            sum_wtlunit(g) = sum_wtlunit(g) + col%wtlunit(c)
```

Comment:
> To handle the possibility that more than one landunit is included here (in case that happens in the future), I think the area weightings need to be changed from col%wtlunit to col%wtgcell. (Then sum_wtlunit should probably be renamed.)

Alternatively, you could do separate renormalizations for each landunit. I haven't thought this through carefully, but I think that would involve accumulating a separate sum for each landunit and doing the renormalization on the landunit level rather than the gridcell level.